files/puppet_helper.groovy

```diff
@@ -947,11 +947,14 @@ class Actions {
     def jobs = util.findJobs(Jenkins.getInstance())
 
     def allInfo = jobs.collect { path, job ->
+      // at least these job classes do not respond to respond to #isDisabled:
+      // - org.jenkinsci.plugins.workflow.job.WorkflowJob
+      // - com.cloudbees.hudson.plugins.folder.Folder
       def enabled = false
-      // folders don't respond to #isDisabled
-      if (job.getClass().getName() != 'com.cloudbees.hudson.plugins.folder.Folder') {
+      if (job.metaClass.respondsTo(job, 'isDisabled')) {
```

Comment:
My first thought, as my eyes focused in on this line: "is this some weird Puppet stuff?" Briefly puzzling to myself "metaClass? respondsTo instead of `respond_to?`? This isn't Ruby!"

Then I looked up and realized this is a Groovy file.


 :coffee: 
 :coffee: 
 :coffee: 
 :coffee: 
 :coffee: 


Carry on